### ISSUE TITLE
Highway label distortion

### ISSUE THREAD

**Issue by kcwu** [osmandapp/Osmand#1756]
For offline map, the label of highway will render along the curve of highway. If the highway is zigzag, e.g. hiking trails, the label is hard to read. For Chinese, this problem is severer.

In this example, the label of top highway is "Hu Shan Zi Ran Bu Dao".
![screenshot_20150930-002635](https://cloud.githubusercontent.com/assets/175566/10171221/8fedbfdc-670b-11e5-8a27-3cc5fd89e4e1.png)

This Chinese label is "虎山自然步道".
![screenshot_20150930-002653](https://cloud.githubusercontent.com/assets/175566/10171269/cde4768c-670b-11e5-83cc-e1329edae647.png)

**Comment by sonora** [comment]
Yes, true. I had played with that some time ago, and found that not labeling along the course of the path makes it difficult to identify what the labels is actually for, as there can be many other labels in the vicinity for nature preserves, hamlets, etc.

From what I experimented, I found it best to live with that as is. For very crooked paths, you need to zoom in further until the bends in the path get reasonably stretched, then you can easily read every label.

**Comment by kcwu** [comment]
Idea to mitigation:
Before rendering the label, "simplify" the path first. I guess we don't need complicated algorithm like Ramer–Douglas–Peucker algorithm, simple rule like "remove nodes if they are too close (say, <50px, depends on font size)" is good enough.

**Comment by c-e-p-x-u-o** [comment]
same for cyrillic font and river
good:
![screenshot_osmand _20180916-024201](https://user-images.githubusercontent.com/14617699/45591402-65bed300-b95a-11e8-8878-9ac0430111e0.png)

absolutely unreadable:
![screenshot_osmand _20180916-024228](https://user-images.githubusercontent.com/14617699/45591405-6eafa480-b95a-11e8-9b33-e4ea4ebdab1e.png)
